### PR TITLE
TradeRoute deployment script v1

### PR DESCRIPTION
**Pull request by nambrot**
This PR makes the deployment script  a bit nicer and fully extracts the configuration necessary to do a TradeRoute deployment.

Fixes https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/1462

### CODE REVIEW

File: trade-route-example-config.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
+  "goerli": {
+    "type": "collateral",
+    "token": "0xb4fbf271143f4fbf7b91a5ded31805e42b2208d6",
+    "owner": "0x5bA371aeA18734Cb7195650aFdfCa4f9251aa513",
+    "mailbox": "0xCC737a94FecaeC165AbCf12dED095BB13F037685",
+    "interchainGasPaymaster": "0xf857706CE59Cb7AE6df81Bbd0B0a656dB3e6beDA"
```
Comment (jmrossy) [inline trade-route-example-config.json:7]: Would be cool if dev didn't have to specify this and it was retrieved from core artifacts instead

File: trade-route-example-config.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  "fuji": {
+    "type": "synthetic",
+    "name": "Dai",
+    "symbol": "DAI",
+    "totalSupply": 0,
+    "token": "0xb4fbf271143f4fbf7b91a5ded31805e42b2208d6",
+    "owner": "0x5bA371aeA18734Cb7195650aFdfCa4f9251aa513",
+    "mailbox": "0xCC737a94FecaeC165AbCf12dED095BB13F037685",
+    "interchainGasPaymaster": "0xf857706CE59Cb7AE6df81Bbd0B0a656dB3e6beDA"
+  },
```
Comment (jmrossy) [inline trade-route-example-config.json:28]: Seems like these configs could be DRY-ed up with other remote configs but nbd

File: hardhat.config.ts [hunk 3]
```diff
@@ -1,8 +1,50 @@
 import '@nomiclabs/hardhat-ethers';
 import '@nomiclabs/hardhat-waffle';
 import '@typechain/hardhat';
+import * as ethers from 'ethers';
```
Comment (jmrossy) [inline hardhat.config.ts:4]: This is a strange syntax for importing from ethers, would normally be just import {ethers} from 'ethers'

**Review by jmrossy — COMMENTED** [review]
mostly looks good!

File: trade-route-example-config.json (hunk 1 above)
Comment (nambrot) [reply trade-route-example-config.json:7]: That's what we had originally before, but the goal here was to support configuration from the user side and JSON seemed preferable to a typescript file

File: trade-route-example-config.json (hunk 2 above)
Comment (nambrot) [reply trade-route-example-config.json:28]: Also was similar before, but again wanted to move to a dependency-less format

File: trade-route-example-config.json (hunk 1 above)
Comment (yorhodes) [reply trade-route-example-config.json:7]: I think @jmrossy 's point is we can pull this from the sdk package

Comment (nambrot) [reply trade-route-example-config.json:7]: Yeah but my intention was for this to be a JSON file and not a Typescript file (the former obviously can't import).

You could make the script just fill in the gaps I guess, but I was (possibly prematurely) anticipating a permissionless deployment world where these contract addresses may not be retrievable from the SDK. I personally think that this is fine enough for folks to handle

**Review by jmrossy — APPROVED** [review]

**Pull request merged**